mne/io/base.py

```diff
@@ -2140,6 +2133,19 @@ def _repr_html_(self):
             duration=duration,
         )
 
+    def _get_duration_timedelta(self):
+        seconds = self.n_times / self.info["sfreq"]
+        return timedelta(seconds=seconds)
+
+    def _get_duration_string(self, duration):
+        # https://stackoverflow.com/a/10981895
+        hours, remainder = divmod(duration.seconds, 3600)
+        minutes, seconds = divmod(remainder, 60)
+        seconds += duration.microseconds / 1e6
+        seconds = np.ceil(seconds)  # always take full seconds
+
+        return f"{int(hours):02d}:{int(minutes):02d}:{int(seconds):02d}"
```

Comment:
I'm in favor of the change to using `self.n_times / self.info["sfreq"]` instead of `self.times[-1]`. But the rest of this seems a bit overcomplicated, I'm not sure why `timedelta` is even needed?  Why not this, either inside `_repr_html_` or as a single helper method?

```py
duration = self.n_times / self.info["sfreq"]
hours, remainder = divmod(duration, 3600)
minutes, seconds = divmod(remainder, 60)
duration = f"{hours:02.0f}:{minutes:02.0f}:{np.ceil(seconds):02.0f}"
```

Reply:
Hi @drammock, I moved the timedelta back into the `._get_duration_string()` method.

On the other hand, I've factored out the `self.n_times / self.info["sfreq"]` calculation into a new `.duration` property.

The `timedelta` usage was originally added to the string representation to use in the rounding-up of the seconds amount, so indeed there is no sense in having a method just to return it.

On the other hand, having a separate `._get_duration_string` method helps testing it in isolation from the `._repr_html_()` method which is barely tested otherwise.

Reply:
And I kept the timedelta on the refactored method, because I didn't want to mess with the logic for calculating `_repr_html_`.

But if you want it removed, I'm ok with that.